content/post/ddos-detection.md

```diff
@@ -0,0 +1,419 @@
+                          server_partitions)
+    ab.to_sink(wallaroo.TCPSinkConfig(out_host, out_port, Encoder()))
+    return ab.build()
+```
+
+## Running the Application
+
+To run the application, we will use netcat for our TCP receiver, and a custom sender that's included in the [demo code][ddos-detection] which will send server metrics in batches of 1000, and attempt to keep the timing roughly in line with the time deltas of the input records by sleeping between batches.
+
+The following steps will run the application:
+
+1. Set up Wallaroo for your OS following [these instructions][setup instructions]
+2. Check out the [Wallaroo Blog Exmaples GitHub repository][wallaroo blog examples] and navigate to the ddos-detection example directory
+
+    ```bash
+    git clone https://github.com/WallarooLabs/wallaroo_blog_examples.git
+    cd wallaroo_blog_examples/ddos-detection
+    ```
+
+3. Start the included listener in its own terminal so you can view the output
+
+    ```bash
+    python receiver.py --host 127.0.0.1:7002
+    ```
+
+4. Start two Wallaroo workers
+    in two separate terminals:
+    1. Initializer:
+
+        ```bash
+        export PYTHONPATH="$PYTHONPATH:.:$HOME/wallaroo-tutorial/wallaroo/machida"
+        export PATH="$PATH:$HOME/wallaroo-tutorial/wallaroo/machida/build"
+        machida --application-module ddos_detector --in 127.0.0.1:7010 --out 127.0.0.1:7002 \
+          --control 127.0.0.1:12500 --data 127.0.0.1:12501 --name initializer --cluster-initializer \
+          --worker-count 2 --metrics 127.0.0.1:5001 --ponythreads 1 --ponypinasio --ponynoblock
+        ```
+
+    2. 2nd Worker:
+
+        ```bash
+        export PYTHONPATH="$PYTHONPATH:.:$HOME/wallaroo-tutorial/wallaroo/machida"
+        export PATH="$PATH:$HOME/wallaroo-tutorial/wallaroo/machida/build"
+        machida --application-module ddos_detector --in 127.0.0.1:7010  --out 127.0.0.1:7003 \
+          --control 127.0.0.1:12500  --name worker --metrics 127.0.0.1:5001 --ponythreads 1 \
+          --ponypinasio --ponynoblock
+        ```
+
+4. Start the sender:
+
+    ```bash
+    python sender.py --host 127.0.0.1:7010 --file data.json --batch 1000
+    ```
+
+![demo gif][demo gif]
+
+## Next Steps
+
+This example builds on the concepts introduced in [Non-native event-driven windowing in Wallaroo][windowing in wallaroo], adding partitioning, parallelization, and the application of statistical analysis to perform a simplified anomaly detection.
+
+However, as this example is designed for illustration purposes, it is not without limitations: the thresholds used for preempting false-positives aren't adaptive, and may not work well with certain loads; the anomaly detection algorithm itself is very simplistic in its assumptions, and may falsely flag legitimate increases in traffic as malicious; likewise, it may fail to detect malicious traffic that isn't large enough relative to the legitimate traffic.
```

Comment:
I think you should finish the thought of this paragraph. Something like

For a real world use case, we would have more sophisticated algos to address these faults.